src/Protocol.h

```diff
@@ -122,16 +122,20 @@ struct HeaderPrefix {
 
 /**
  * Describes the wire format for header fields that are common to all packet
- * types.
+ * types. Note: the first 4 bytes are identical for TCP, UDP, and Homa.
  */
 struct CommonHeader {
+    uint16_t sport, dport;///< Transport layer (L4) source and destination ports
+                          ///< in network byte order; only used by DataHeader.
     HeaderPrefix prefix;  ///< Common to all versions of the protocol.
```

Comment:
The `prefix` should be the first element of the struct. It's used for future protocol versioning purposes and we should always know where to find it no matter the version of the protocol is running.

Reply:
This is necessary to be consistent with TCP and UDP. Shenango's network stack assumes that the first 4 bytes of L4 protocols will be the same.